sklearn/neighbors/lof.py

```diff
@@ -308,20 +356,29 @@ def _score_samples(self, X):
             The opposite of the Local Outlier Factor of each input samples.
             The lower, the more abnormal.
         """
-        check_is_fitted(self, ["offset_", "negative_outlier_factor_",
-                               "_distances_fit_X_"])
-        X = check_array(X, accept_sparse='csr')
 
-        distances_X, neighbors_indices_X = (
-            self.kneighbors(X, n_neighbors=self.n_neighbors_))
-        X_lrd = self._local_reachability_density(distances_X,
-                                                 neighbors_indices_X)
+        if self.novelty:
+            check_is_fitted(self, ["offset_", "negative_outlier_factor_",
+                                   "_distances_fit_X_"])
+            X = check_array(X, accept_sparse='csr')
+
+            distances_X, neighbors_indices_X = (
+                self.kneighbors(X, n_neighbors=self.n_neighbors_))
+            X_lrd = self._local_reachability_density(distances_X,
+                                                     neighbors_indices_X)
+
+            lrd_ratios_array = (self._lrd[neighbors_indices_X] /
+                                X_lrd[:, np.newaxis])
 
-        lrd_ratios_array = (self._lrd[neighbors_indices_X] /
-                            X_lrd[:, np.newaxis])
+            # as bigger is better:
+            return -np.mean(lrd_ratios_array, axis=1)
 
-        # as bigger is better:
-        return -np.mean(lrd_ratios_array, axis=1)
+        msg = ('score_samples is not available when novelty=False. The '
+               'score_samples on training data is always available by '
+               'considering the negative_outlier_factor_ attribute. Use '
+               'novelty=True if you want to use LOF for novelty detection and '
+               'compute score_samples for new unseen data.')
+        raise NotImplementedError(msg)
```

Comment:
I would structure the code differently.

```
if not self.novelty:
     raise

...

return ...```

it feels better to first capture errors and to end the function with a return